Support MongoDB connection type for AWS Glue

drexler opened the pull request:
<!--- See what makes a good Pull Request at : https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#pull-requests --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Closes #13007

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
<!--
If change is not user facing, just write "NONE" in the release-note block below.
-->

```release-note
support MongoDB connection type for AWS Glue
```

Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueConnection_MongoDB'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueConnection_MongoDB -timeout 120m
=== RUN   TestAccAWSGlueConnection_MongoDB
=== PAUSE TestAccAWSGlueConnection_MongoDB
=== CONT  TestAccAWSGlueConnection_MongoDB
--- PASS: TestAccAWSGlueConnection_MongoDB (15.14s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	15.208s


...
```

anGie44 commented on aws/resource_aws_glue_connection_test.go:
```diff
@@ -439,3 +472,19 @@ resource "aws_glue_connection" "test" {
+
+func testAccAWSGlueConnectionConfig_MongoDB(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_glue_connection" "test" {
```
small nit-pick here to align this config with `terrafmt fmt`

drexler commented:
Thanks for the review @anGie44. I've updated the PR and added the documentation snippet.

>I think it'd be helpful to document the newly supported connection_type in glue_connection.html.markdown (e.g. "Supported: JDBC, SFTP, MONGODB")

Interestingly enough SFTP is [**not supported with the Go SDK**](https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#ConnectionInput) so omitted that from the documentation. Should i file a follow-up issue to remove it from the resource validation check? Alternatively, i could do it as part of this PR - it doesn't seem to be a breaking change to do.

anGie44 commented:
@drexler interesting! hmm seems like the online API docs haven't been updated.. yes, please file a follow-up issue so we have it documented. thanks for investigating :)

drexler commented:
@anGie44 issue for SFTP connection support created.
